added flipDirection/flipHour settings, stopname and filterDirection fix

gmichaela opened the pull request:
Fixes #5.

added the filterDirection #5 filter, upon doing so, realized I care more about one direction in the morning and a different direction in the afternoon #6? -- added a few more config flags, if flipDirection is set to true it will switch the direction at noon, if you want an hour other than noon as your flip time -- override with flipHour.

few other minor edits/fixes also included.

gmichaela committed:
added flipDirection/flipHour settings, stopname and filterDirection fix

edward-shen commented on MMM-MBTA.js:
```diff
@@ -235,7 +251,7 @@ Module.register("MMM-MBTA", {
                     var preETACell = document.createElement("td");
                     var preETATime = this.stationData[i].preETA;
-
+                    
```
Please remove trailing whitespace.
```suggestion

```

edward-shen commented on MMM-MBTA.js:
```diff
@@ -1,11 +1,11 @@
     defaults: {
         apikey: "",
-        updateInterval: 10, // In seconds
+        updateInterval: 30, // In seconds
```
Any particular reason for the change?

edward-shen commented on MMM-MBTA.js:
```diff
@@ -20,8 +20,11 @@ Module.register("MMM-MBTA", {
+        hideEmptyAlerts: false,
+        flipDirection: false,  // if set to true, it will flip direction filter flag,
+        flipHour: 12, 
+        directionFlipped: false
```
For all these values, can you please update the readme with these flags? Following what's been done is good enough!

edward-shen commented on MMM-MBTA.js:
```diff
@@ -124,7 +128,7 @@ Module.register("MMM-MBTA", {
 
             var descCell = document.createElement("td");
-            descCell.innerHTML = "No vehicles are scheduled";
+            descCell.innerHTML = "Nothing coming soon";
```
Missing a period or a "..."

```suggestion
            descCell.innerHTML = "Nothing coming soon...";
```

edward-shen commented on MMM-MBTA.js:
```diff
@@ -436,9 +452,9 @@ Module.register("MMM-MBTA", {
         // Page and time limits necessary because otherwise "schedules" endpoint gets every single schedule
         if (!this.config.predictedTimes) {
-            url += "&page[limit]=" + (maxEntries + 10) + '"';
+            url += "&page[limit]=" + (this.config.maxEntries + 10) + '"';
```
Nice catch!

edward-shen commented on MMM-MBTA.js:
```diff
@@ -436,9 +452,9 @@ Module.register("MMM-MBTA", {
+            url += "&page[limit]=" + (this.config.maxEntries + 10) + '"';
             url += "&filter[min_time]=" + moment().format("HH:mm");
-            url += "&filter[max_time]=" + moment().add(3, 'h').format("HH:mm");
+            url += "&filter[max_time]=" + moment().add(5, 'h').format("HH:mm");
```
Any particular reason for this change? if you want to keep it at five, make sure to update the comment on line 452!

gmichaela replied:
I wanted to see more trips coming, commuter rails don't have too many options in the 3 hour window sometimes...   the predictedTimes api doesnt seem to return too far in advance either, so I was just experimenting with adding more time here -- I saw that you truncated the list at the end anyway...

edward-shen replied:
Alright, that's understandable! Just update the comment and you should be all set for this part!

edward-shen commented on MMM-MBTA.js:
```diff
@@ -613,6 +631,34 @@ Module.register("MMM-MBTA", {
         this.stationData.sort((a,b) => (a.preETA - b.preETA));
 
+        if (this.config.flipDirection && this.config.direction.length > 0)
+        {
```
Please be consistent with the already existing guide. This should be

```
if (...) {
  ...
}
```

Thanks!

edward-shen replied:
There are a few more in the follow code too!

edward-shen commented on MMM-MBTA.js:
```diff
@@ -613,6 +631,34 @@ Module.register("MMM-MBTA", {
+        //  Applys directional filter
+        if (this.filterDirection.length > 0)
+        {
+            this.stationData = this.stationData.filter(obj => (obj.directionId == this.filterDirection));
```
Double equals is discouraged. This should be `===`.
```suggestion
            this.stationData = this.stationData.filter((obj) => obj.directionId === this.filterDirection);
```

gmichaela replied:
ended up it should have been =

edward-shen commented on MMM-MBTA.js:
```diff
@@ -198,8 +202,20 @@ Module.register("MMM-MBTA", {
+                        break;
+                }
 
+
```
Please leave only one blank lines between one thing and the next.

edward-shen reviewed (changes requested):
Looks like a great PR! I just have some minor requests and then it looks good!

edward-shen commented:
Fixes #5.

gmichaela committed:
Update MMM-MBTA.js

Code doesn't allow any faster than 10, api recommendation also no less than 10.

edward-shen committed:
Update MMM-MBTA.js

Co-Authored-By: gmichaela <mamante@yahoo.com>

edward-shen committed:
Update MMM-MBTA.js

Co-Authored-By: gmichaela <mamante@yahoo.com>

edward-shen committed:
Update MMM-MBTA.js

Co-Authored-By: gmichaela <mamante@yahoo.com>

gmichaela committed:
style points and readme info

gmichaela commented:
Made some more edits -- updated readme, fixed a filter direction bug.

edward-shen commented on readme.md:
```diff
@@ -67,12 +67,17 @@ Option|Description
 `hideEmptyAlerts`|Automatically hide the alerts section when there are no alerts.<br/>**Expected Value type:** `boolean`.
+`direction`| Sets the default direction you care about.<br/>Example: `direction:  "Inbound" `.<br/>**Expected Value type:** `String`.
+`flipDirection`| Set to true, it will flip direction filter flag, ie: Inbound in the AM Outbound in PM<br/>**Expected Value type:** `boolean`.
+`flipHour`| When set with flipDirection the time of day you want to flip direction to override 12 noon<br/>**Expected Value type:** `int`.
```
```suggestion
`flipHour`| Specifies the hour to flip the direction, if `flipDirection` is equal to `true`. Default value is 12.**Expected Value type:** `int`.
```

gmichaela replied:
Adding Outbound etc wouldn't hurt either...

edward-shen commented on readme.md:
```diff
@@ -67,12 +67,17 @@ Option|Description
 `showAlerts`|Shows alerts near the specified station.<br/>**Expected Value type:** `boolean`.<br/>**WARNING**: Alerts are only properly shown when in a full-width position, such as `top_bar`, `middle_center` or `lower_third`.
 `hideEmptyAlerts`|Automatically hide the alerts section when there are no alerts.<br/>**Expected Value type:** `boolean`.
+`direction`| Sets the default direction you care about.<br/>Example: `direction:  "Inbound" `.<br/>**Expected Value type:** `String`.
+`flipDirection`| Set to true, it will flip direction filter flag, ie: Inbound in the AM Outbound in PM<br/>**Expected Value type:** `boolean`.
```
```suggestion
`flipDirection`| When set to `true`, MMM-MBTA will the flip the direction filter for incoming vehicles at the station, i.e. Inbound in the AM and Outbound in PM<br/>**Expected Value type:** `boolean`.
```

edward-shen reviewed (approved):
Please accept the suggestions and your PR will be merged! 

Thanks for working with me on this, I really appreciate the PR!

edward-shen committed:
Update readme.md

Co-Authored-By: gmichaela <mamante@yahoo.com>

edward-shen committed:
Update readme.md

Co-Authored-By: gmichaela <mamante@yahoo.com>

edward-shen commented:
Merged! :tada:
